platform/web/js/libs/library_godot_javascript_singleton.js

```diff
@@ -81,11 +81,11 @@ const GodotJSWrapper = {
 			case 0:
 				return null;
 			case 1:
-				return !!GodotRuntime.getHeapValue(val, 'i64');
+				return Boolean(GodotRuntime.getHeapValue(val, 'i64'));
 			case 2:
-				return GodotRuntime.getHeapValue(val, 'i64');
+				return Number(GodotRuntime.getHeapValue(val, 'i64'));
```

Comment:
I wonder if we should add some logic here to keep the value as bigint if it doesn't fit:
```suggestion
				const big = GodotRuntime.getHeapValue(val, 'i64');
				return big > Number.MAX_SAFE_INTEGER || big < Number.MIN_SAFE_INTEGER ? big : Number(big);
```

This might break some code on the JS side (if it only expects numbers), but at least will not result in surprising behaviors (i.e. mismatching values between Godot and JS).

Reply:
Let's try this!